### PR TITLE
docs: lint the markdown docs by markdownlint

### PR DESCRIPTION
**Pull request by ALiangLiang**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

I think our docs is hard to read. And too many unnecessary continuously blank lines 

like this below. It's hard to know the scope of cope block.

~~~~
```
code1 here
```
content here
```
code2 here
```
~~~~

So I follow markdownlint, add a blank line between them and fill in the type on the code block.

~~~~
```js
code1 here
```

content here

```text
code2 here
```
~~~~

---

BTW. I want to discuss should we introduce [markdownlint](https://github.com/DavidAnson/markdownlint) like `eslint`?

### CODE REVIEW

File: docs/hooks.md [hunk 1]
```diff
@@ -244,7 +248,7 @@ User.create({username: 'Boss', accessLevel: 20}).then(user => {
 
 Sometimes you'll be editing more than one record at a time by utilizing the `bulkCreate, update, destroy` methods on the model. The following will emit whenever you're using one of those methods:
 
-```
+```js
```
Comment (SimonSchick) [inline docs/hooks.md]: This really isn't valid JS.

File: docs/whos-using.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Who's using sequelize?
```
Comment (papb) [inline docs/whos-using.md]: I personally didn't like the removal of this question mark... Was this automatically suggested by `markdownlint`?

**Comment by papb** [comment]
As for your question, I personally think it would be nice to have `markdownlint` configured. But not in this PR, I think it would be better to have a different PR for it, where we can discuss precisely which options would be enabled/disabled.

File: docs/whos-using.md (hunk 2 above)
Comment (ALiangLiang) [reply docs/whos-using.md]: Yes. It's a [rule](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md026) `Trailing punctuation in heading`.

Comment (ALiangLiang) [reply docs/whos-using.md]: But I prefer to follow the rule. No question mark is more like a title.

Comment (papb) [reply docs/whos-using.md]: I see... I would personally suggest changing the default config of markdownlint to allow question marks. It's even an example in the docs you linked to. But if you really disagree, at least I think the whole title should be changed then. "Who's using sequelize" without a question mark seems odd to me. Would you suggest a different title?

Comment (ALiangLiang) [reply docs/whos-using.md]: Um..... I was persuaded. Add the question mark back.

**Comment by ALiangLiang** [comment]
@papb Good idea. Later I will create a PR about introducing markdownlint.

**Comment by sushantdhiman** [comment]
@ALiangLiang yeah markdownlint is a good idea

**Comment by ALiangLiang** [comment]
@papb 
I was created the PR. #10692

**Pull request merged**